src/it/java/teammates/it/ui/webapi/FeedbackSessionOpeningRemindersActionIT.java

```diff
@@ -0,0 +1,172 @@
+package teammates.it.ui.webapi;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import teammates.common.util.Const;
+import teammates.common.util.EmailType;
+import teammates.common.util.EmailWrapper;
+import teammates.common.util.HibernateUtil;
+import teammates.common.util.TaskWrapper;
+import teammates.storage.sqlentity.Course;
+import teammates.storage.sqlentity.FeedbackQuestion;
+import teammates.storage.sqlentity.FeedbackSession;
+import teammates.ui.output.MessageOutput;
+import teammates.ui.request.SendEmailRequest;
+import teammates.ui.webapi.FeedbackSessionOpeningRemindersAction;
+import teammates.ui.webapi.JsonResult;
+
+/**
+ * SUT: {@link FeedbackSessionOpeningRemindersAction}.
+ */
+public class FeedbackSessionOpeningRemindersActionIT extends BaseActionIT<FeedbackSessionOpeningRemindersAction> {
+
+    @Override
+    @BeforeMethod
+    protected void setUp() throws Exception {
+        super.setUp();
+        persistDataBundle(typicalBundle);
+        HibernateUtil.flushSession();
+        prepareSession();
+    }
+
+    private void prepareSession() {
+        // FEEDBACK QUESTIONS
+        String[] fqKeys = {
+                "qn1InSession1InCourse1",
+                "qn2InSession1InCourse1",
+                "qn3InSession1InCourse1",
+                "qn4InSession1InCourse1",
+                "qn5InSession1InCourse1",
+                "qn6InSession1InCourse1NoResponses",
+        };
+        List<FeedbackQuestion> qns = new ArrayList<>();
+        for (String fqKey : fqKeys) {
+            qns.add(typicalBundle.feedbackQuestions.get(fqKey));
+        }
+
+        FeedbackSession session = typicalBundle.feedbackSessions.get("session1InCourse1");
+        session.setFeedbackQuestions(qns);
+    }
+
+    @Override
+    String getActionUri() {
+        return Const.CronJobURIs.AUTOMATED_FEEDBACK_OPENING_REMINDERS;
+    }
+
+    @Override
+    String getRequestMethod() {
+        return GET;
+    }
+
+    @Test
+    @Override
+    protected void testAccessControl() throws Exception {
+        Course course = typicalBundle.courses.get("course1");
+        verifyOnlyAdminCanAccess(course);
+    }
+
+    @Test
+    @Override
+    protected void testExecute() throws Exception {
+        loginAsAdmin();
+
+        ______TS("Typical Success Case 1: Email tasks added for co-owner, students and instructors of 1 session");
+        testExecute_typicalSuccess1();
+
+        ______TS("Typical Success Case 2: No email tasks added for session -- already sent opening emails");
+        testExecute_typicalSuccess2();
+
+        ______TS("Typical Success Case 3: No email tasks added for session -- session not visible yet");
```

Comment:
Is the primary reason for this not because the session is not open yet? I do not think we did a check for session visibility. (I don't know if we should though.)

Reply:
The visibility check is done in `isFeedbackSessionForUserTypeToAnswer`;

But I think it would be good to add another test for session is not open.